File: @capacitor-community/screenshots/ios/Plugin/ScreenshotsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ScreenshotsPlugin, "Screenshots",
           CAP_PLUGIN_METHOD(disableScreenshots, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(enableScreenshots, CAPPluginReturnPromise);
)

File: @capacitor-community/screenshots/ios/Plugin/ScreenshotsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ScreenshotsPlugin)
public class ScreenshotsPlugin: CAPPlugin {

    @objc func disableScreenshots(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.bridge?.webView?.disableScreenshots()
        }
        call.resolve()
    }

    @objc func enableScreenshots(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.bridge?.webView?.enableScreenshots()
        }
        call.resolve()
    }
}
